nearcore/src/migrations.rs

```diff
@@ -478,6 +479,28 @@ pub fn migrate_24_to_25(path: &Path) {
     set_store_version(&store, 25);
 }
 
+/// Fix an issue with block ordinal (#5761)
+pub fn migrate_30_to_31(path: &Path, near_config: &NearConfig) {
+    let store = create_store(path);
+    if near_config.client_config.archive && &near_config.genesis.config.chain_id == "mainnet" {
```

Comment:
The fact that we're running it "only on mainnet" - means that this code will not be executed in testnet etc.

What I'd do is:
 - remove this condition (so that code runs on testnet / betanet etc)
 - in 491 -> say that for "mainnet" we start from block 47443088, while for other networks, we go from block 1

Reply:
> in 491 -> say that for "mainnet" we start from block 47443088, while for other networks, we go from block 1

I think this is debatable. Given that the check on mainnet (going from genesis block) took about 27 hours, I am not sure whether we want to do it on testnet as that will incur a significant amount of downtime for an archival node. At the same time, I manually tested this code on a mainnet archival node and it works fine (confirmed by Aurora).

@mfornet could you let us know whether it is important to run migration for testnet as well?

Reply:
So far we haven't seen this issue on Testnet, but it is hard to know if the issue is present for some blocks in the past.
If it is the case it doesn't affect future txs, so from the POV of the bridge, running the migration in testnet is not necessary.

Reply:
@bowenwang1996 Yes, we haven't experienced that issue on Testnet so far, therefore, I don't think this is required to run a migration there.

N.B.: I will let you know if we encounter this issue there.